feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/moods/addNewMoodActivity.java

```diff
@@ -33,15 +47,20 @@
 
 import io.opencensus.internal.StringUtils;
 
+import static com.google.android.gms.common.internal.safeparcel.SafeParcelable.NULL;
+
 /*Responsible for editing or adding a mood event*/
 
 public class addNewMoodActivity extends AppCompatActivity {
 
+    public static final int MY_PERMISSIONS_REQUEST_LOCATION = 99;
     private Spinner moodSpiner;
     private Spinner socialSituationSpinner;
+    private Switch locationSwitch;
+    private boolean saveLocation = false;
     private EditText reason;
     public User currentUser;
-    private int moodState = 0; //if this variable =0 it means you're addind a new mood, it 1 you're editing the current mood
+    private int moodState = 0; //if this variable =0 it means you're adding a new mood, it 1 you're editing the current mood
```

Comment:
I know this isn't your code per se, but do we want to consider changing this to a Boolean?